vital/range/iterator_range.h

```diff
@@ -29,6 +32,10 @@ class iterator_range
   Iterator begin() const { return m_begin; }
   Iterator end() const { return m_end; }
 
+  bool empty() const { return !( m_begin != m_end ); }
+
+  size_t size() const { return std::distance( m_begin, m_end ); }
```

Comment:
```suggestion
  // TODO(C++14): deduce return type and remove <cstddef>
  size_t size() const { return std::distance( m_begin, m_end ); }
```

Even without C++14, I'd be tempted to use a trailing return specification here and avoid `cstddef`. (With a `// TODO(C++14) use pure return type deduction` 😛.)